posts/2022-12-15-Rust-1.66.0.md

```diff
@@ -0,0 +1,171 @@
+---
+layout: post
+title: "Announcing Rust 1.66.0"
+author: The Rust Release Team
+release: true
+---
+
+The Rust team is happy to announce a new version of Rust, 1.66.0. Rust is a
+programming language empowering everyone to build reliable and efficient
+software.
+
+If you have a previous version of Rust installed via rustup, you can get 1.66.0
+with:
+
+```console
+rustup update stable
+```
+
+If you don't have it already, you can [get
+`rustup`](https://www.rust-lang.org/install.html) from the appropriate page on
+our website, and check out the [detailed release notes for
+1.66.0](https://github.com/rust-lang/rust/blob/stable/RELEASES.md#version-1660-2022-12-15)
+on GitHub.
+
+If you'd like to help us out by testing future releases, you might consider
+updating locally to use the beta channel (`rustup default beta`) or the nightly
+channel (`rustup default nightly`). Please
+[report](https://github.com/rust-lang/rust/issues/new/choose) any bugs you
+might come across!
+
+## What's in 1.66.0 stable
+
+### Explicit discriminants on enums with fields
+
+Enums with integer representations can now use explicit discriminants, even when they have fields.
+
+```rust
+#[repr(u8)]
+enum Foo {
+    A(u8),
+    B(i8),
+    C(bool) = 42,
+}
+```
+
+Previously, you could use explicit discriminants on enums with representations, but only if none of their variants had fields. Explicit discriminants are useful when passing values across language boundaries where the representation of the enum needs to match in both languages. For example,
+
+```rust
+#[repr(u8)]
+enum Bar {
+    A,
+    B,
+    C = 42,
+}
+```
+
+Here the `Foo` enum is guaranteed to have the same layout as `u8`. Each variant will use either the specified discriminant value or default to starting with 0.
+
+```rust
+assert_eq!(0, Bar::A as u8);
+assert_eq!(1, Bar::B as u8);
+assert_eq!(42, Bar::C as u8);
+```
+
+You could even add fields to enums with `#[repr(Int)]`, and they would be laid out in a predictable way. Previously, however, you could not use these features together. That meant that making `Foo::C`'s discriminant equal to 42 as above would be harder to achieve. You would need to add 41 hidden variants in between as a workaround with implicit discriminants!
+
+Starting in Rust 1.66.0, the above example compiles, allowing you to use explicit discriminants on any enum with a `#[repr(Int)]` attribute.
+
+### `core::hint::black_box`
+
+When benchmarking or examining the machine code produced by a compiler, it's often useful to prevent optimizations from ocurring in certain places. In the following example, the function `push_cap` executes `Vec::push` 4 times in a loop:
```

Comment:
```suggestion
When benchmarking or examining the machine code produced by a compiler, it's often useful to prevent optimizations from occurring in certain places. In the following example, the function `push_cap` executes `Vec::push` 4 times in a loop:
```